Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -275,6 +275,47 @@ def check_python_symlinks(name, keg_only)
     "Python formulae that are keg-only should not create `pip3` and `wheel3` symlinks."
   end
 
+  INVALID_EMPTY_FILE_NAMES = %w[AUTHORS ChangeLog changes COPYING NEWS README TODO].freeze
+  INVALID_EMPTY_FILE_EXTENSIONS = %w[.html .md .rst .txt].freeze
+  INVALID_EMPTY_FILE_LOCATIONS = %w[bin libexec share/man share/doc].freeze
+  # Names and extensions to ignore, even if they are in a location within prefix given above
+  VALID_EMPTY_FILE_NAMES = %w[__init__.py REQUESTED].freeze
+  VALID_EMPTY_FILE_EXTENSIONS = %w[.stamp].freeze
+
+  def check_empty_files(prefix)
+    return unless prefix.directory?
+
+    empty_files = []
+    prefix.children.each do |f|
+      next if INVALID_EMPTY_FILE_NAMES.none?(File.basename(f)) &&
+              INVALID_EMPTY_FILE_EXTENSIONS.none?(File.extname(f))
```

Comment:
```suggestion
      next if INVALID_EMPTY_FILE_NAMES.include?(f.basename)
      next if INVALID_EMPTY_FILE_EXTENSIONS.include?(f.extname)
```

Reply:
This suggestion would create different behavior. My intended behavior was to find files that matched **either** an invalid name or invalid extension.

Example expected behavior:
Searched files include `TODO`, `TODO.md`, and `asdf.md`
Files that aren't searched include `iojaoisjv`
(provided the above are all directly in the `#{prefix}`, not a subdirectory)

With this change:
Only `TODO` and `iojaoisjv` (!!) are found, the rest aren't looked at

Then when I revert `f.basename` and `f.extname` to `File.basename(f)` and `File.extname(f)`., `TODO` is no longer found and `iojaoisjv` is the only one that is found (basically, the exact opposite scenario of what I want).

Through some trial and error and print statements, I found that `f.basename` outputs names in the form `NAMEANDEXT/`, while File.basename(f) outputs them in the form `NAMEANDEXT`, so then I used `f.basename(".*").to_s.delete_suffix("/")` to remove the extension and the "/".

But then the `next if`s still have the opposite behavior from what I want, and if I change them both to `next unless`, then the files are only checked if they match both the name and the extension (which isn't what I want either). I'm not sure how to do this without a logical operator `||`, something like:

```
next unless INVALID_EMPTY_FILE_NAMES.include?(f.basename(".*").to_s.delete_suffix("/")) ||
            INVALID_EMPTY_FILE_EXTENSIONS.include?(f.extname)
```

Any other suggestions?

Reply:
> This suggestion would create different behavior. My intended behavior was to find files that matched **either** an invalid name or invalid extension.

Gotcha, I misunderstood.

Use `next if .exclude? && .exclude?`.